File: desktop/windows/src/renderer/src/lib/shareLinks.ts
/** Public share-link base URL for self-hosting (#4339). Matches backend OMI_SHARE_BASE_URL. */

import { resolveWindowsDeployment } from '../../../shared/deploymentProfile'

const DEFAULT_SHARE_BASE = 'https://h.omi.me'

export function shareBaseUrl(
  raw: string | undefined = import.meta.env.VITE_OMI_SHARE_BASE_URL as string | undefined
): string {
  const deployment = resolveWindowsDeployment()
  let value = (raw ?? '').trim()
  if (!value) value = deployment.shareBase ?? ''
  if (!value) throw new Error('Public conversation sharing is disabled by this deployment profile')
  if (!value.includes('://')) value = `https://${value}`
  try {
    const parsed = new URL(value)
    if (!['http:', 'https:'].includes(parsed.protocol) || !parsed.hostname) {
      if (deployment.profile === 'self_hosted') throw new Error('Invalid self-hosted share origin')
      return DEFAULT_SHARE_BASE
    }
  } catch {
    if (deployment.profile === 'self_hosted') throw new Error('Invalid self-hosted share origin')
    return DEFAULT_SHARE_BASE
  }
  return value.replace(/\/+$/, '')
}

export function conversationShareUrl(
  id: string,
  raw?: string | undefined
): string {
  return `${shareBaseUrl(raw)}/conversations/${id}`
}

File: app/lib/utils/share_links.dart
// Public share-link base URL for self-hosting (#4339).
//
// Matches backend `OMI_SHARE_BASE_URL` / desktop share helpers.
// Override at build time with `--dart-define=OMI_SHARE_BASE_URL=https://share.example.com`.

import 'package:omi/env/env.dart';
import 'package:omi/env/environment_profile.dart';

const defaultShareBaseUrl = 'https://h.omi.me';

/// Return the configured share origin (no trailing slash).
///
/// [raw] is for tests; production callers omit it so the dart-define / default apply.
String shareBaseUrl([String? raw]) => shareBaseUrlForProfile(raw, Env.profile);

String shareBaseUrlForProfile(String? raw, AppEnvironmentProfile configuredProfile) => Env.resolveShareBaseUrl(
      configuredProfile: configuredProfile,
      configuredShareUrl: raw,
    );

/// Join [shareBaseUrl] with a path (leading slash optional).
String buildShareUrl(String path, {String? raw}) {
  final normalized = path.startsWith('/') ? path : '/$path';
  return '${shareBaseUrl(raw)}$normalized';
}

String conversationShareUrl(String conversationId, {String? raw}) =>
    buildShareUrl('/conversations/$conversationId', raw: raw);

String appShareUrl(String appId, {String? raw}) => buildShareUrl('/apps/$appId', raw: raw);

String recapShareUrl(String summaryId, {String? raw}) => buildShareUrl('/recaps/$summaryId', raw: raw);
